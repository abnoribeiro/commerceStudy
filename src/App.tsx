import React from 'react';
import { StatusBar } from 'react-native';
import { NavigationContainer } from '@react-navigation/native';
import { ThemeProvider } from 'styled-components';
import { 
  useFonts,
  Ruda_400Regular,
  Ruda_500Medium,
  Ruda_600SemiBold,
  Ruda_700Bold
} from '@expo-google-fonts/ruda';

import {Container} from './styled';
import GlobalContext from './hooks';
import { AppRoutes } from './routes/app.routes';
import theme from './_theme/themes';
import themes from './_theme/themes';

const App: React.FC = () => {
  
  const [fontsLoaded] = useFonts({
    Ruda_400Regular,
    Ruda_500Medium,
    Ruda_600SemiBold,
    Ruda_700Bold
  })
  
  if(!fontsLoaded){
    return (<></>)
  }

  return (
    <GlobalContext>
      <StatusBar
        hidden = {false}
      />
      <ThemeProvider theme={themes}>
        <Container>
          <NavigationContainer>
            <AppRoutes/>
          </NavigationContainer>
        </Container>
      </ThemeProvider>
    </GlobalContext>
  );
};

export default App;