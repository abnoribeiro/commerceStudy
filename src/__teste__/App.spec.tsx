import React from "react";
import render from 'react-test-renderer';
import App from "../App";

test('Check render App', ()=>{
  const AppRender = render.create(<App/>).toJSON();
  expect(AppRender).toMatchSnapshot();
});