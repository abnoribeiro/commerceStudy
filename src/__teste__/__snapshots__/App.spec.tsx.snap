// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check render App 1`] = `null`;
